### PR TITLE
MVKQueue: Make sure waitIdle waits for all commands to be encoded before enqueuing its own waitUntilCompleted.

### PR DESCRIPTION
**Pull request by js6i**
That seems to be causing mysterious crashes with asynchronous queue submissions. I've been debugging one where it'd crash the encoding thread because we destroyed the swap chain right after calling `vkQueueWaitIdle`, and it wasn't done encoding the Present.

For what it's worth in my case the bug bisected to f0cb31a12b59f05177f07ab5a46bc9084ba5fbc9.

Ping #2295, possibly others..

### CODE REVIEW

**Comment by js6i** [comment]
#2031 looks possibly related.

**Review by billhollings — APPROVED** [review]
Very nice catch! Thanks for submitting this!

LGTM.

Just going to give @cdavis5e a chance to chime in, before pulling this in.

File: MoltenVK/MoltenVK/GPUObjects/MVKQueue.mm [hunk 1]
```diff
@@ -85,7 +85,16 @@
...
-		dispatch_async(_execQueue, ^{ execute(qSubmit); } );
+		std::unique_lock lock(_execQueueMutex);
+		_execQueueJobCount++;
+
+		dispatch_async(_execQueue, ^{
+			execute(qSubmit);
+
+			std::unique_lock execLock(_execQueueMutex);
+			_execQueueJobCount--;
+			_execQueueConditionVariable.notify_all();
```
Comment (cdavis5e) [inline MoltenVK/MoltenVK/GPUObjects/MVKQueue.mm:96]: The wait code below checks that the job count is zero before actually exiting the wait. You could probably avoid spurious wakeups by only notifying when `_execQueueJobCount` becomes zero. (Or did you try that and experience problems?)

**Review by cdavis5e — APPROVED** [review]
One small optimization I noticed.

File: MoltenVK/MoltenVK/GPUObjects/MVKQueue.mm (hunk 1 above)
Comment (js6i) [reply MoltenVK/MoltenVK/GPUObjects/MVKQueue.mm]: No problems, I think that's good.

**Pull request merged**